packages/lab/src/pill-next/PillNext.tsx

```diff
@@ -1,67 +1,61 @@
 import pillCss from "./PillNext.css";
-import { CloseIcon } from "@salt-ds/icons";
-
-export interface PillNextProps extends ComponentPropsWithoutRef<"button"> {
-  /* If true the pill will be disabled */
-  disabled?: boolean;
-  onClose?: (event: MouseEvent<HTMLButtonElement>) => void;
-  /* Pass an element to render an icon descriptor on the left of the label */
-  icon?: React.ReactNode;
-}
 
 const withBaseName = makePrefixer("saltPillNext");
 
-export const PillNext = forwardRef<HTMLButtonElement, PillNextProps>(
-  function PillNext(
-    { children, className, icon, disabled, onClose, ...restProps },
-    ref
-  ) {
-    const targetWindow = useWindow();
-    useComponentCssInjection({
-      testId: "salt-pill-next",
-      css: pillCss,
-      window: targetWindow,
-    });
-    const { buttonProps, active } = useButton<HTMLButtonElement>({
-      disabled,
-      ...restProps,
-    });
-    // we do not want to spread tab index in this case because the button element
-    // does not require tabindex="0" attribute
-    // eslint-disable-next-line @typescript-eslint/no-unused-vars
-    const { tabIndex, ...restButtonProps } = buttonProps;
-
-    return (
-      <div className={clsx(withBaseName(), className)}>
-        <button
-          data-testid="pill"
-          ref={ref}
-          className={clsx(withBaseName("action"), withBaseName("clickable"), {
-            [withBaseName("active")]: active,
-            [withBaseName("disabled")]: disabled,
-          })}
-          {...restButtonProps}
-          {...restProps}
-        >
-          {icon}
-          <span className={withBaseName("label")}>{children}</span>
-        </button>
-        {onClose && (
-          <Button
-            data-testid="pill-close-button"
-            className={withBaseName("close-button")}
-            disabled={disabled}
-            onClick={onClose}
-          >
-            <CloseIcon />
-          </Button>
-        )}
-      </div>
-    );
-  }
-);
+export const PillNext = forwardRef<
+  HTMLButtonElement,
+  ComponentPropsWithoutRef<"button">
+>(function PillNext(
+  {
+    children,
+    className,
+    disabled,
+    onKeyUp,
+    onKeyDown,
+    onClick,
+    onBlur,
+    ...rest
+  },
+  ref
+) {
+  const targetWindow = useWindow();
+  useComponentCssInjection({
+    testId: "salt-pill-next",
+    css: pillCss,
+    window: targetWindow,
+  });
+  const { buttonProps, active } = useButton<HTMLButtonElement>({
+    disabled,
+    onKeyUp,
+    onKeyDown,
+    onClick,
+    onBlur,
+  });
+  // we do not want to spread tab index in this case because the button element
+  // does not require tabindex="0" attribute
+  // eslint-disable-next-line @typescript-eslint/no-unused-vars
+  const { tabIndex, ...restButtonProps } = buttonProps;
+  return (
+    <button
+      data-testid="pill"
+      ref={ref}
+      className={clsx(
+        withBaseName(),
+        {
+          [withBaseName("clickable")]: Boolean(onClick),
```

Comment:
You've added a new state here, but we're now missing tests for it and examples. A non-clickable Pill

Reply:
hahaha features i add when im trying to add bugs, should we keep unclickable pill or do i move it back to always clickable?

Reply:
To simplify this PR I would keep it always clickable for now